app/validation/index.js

```diff
@@ -43,12 +43,46 @@ class Validator {
     if (isEmpty(data.email)) {
       this.errors.emailEmpty = 'Please provide your email address.';
     }
+    if (!isEmail(data.email)) {
+      this.errors.emailInvalid = 'Please provide a valid email address.';
+    }
+    if (!isValid(data.password)) {
+      this.errors.passwordInputInvalid = 'The provided input for the password is not valid.';
+    }
+    if (isEmpty(data.password)) {
+      this.errors.passwordEmpty = 'Please provide your password.';
+    }
+    return this.checkForErrors();
```

Comment:
I think it would be best to have one error field, then progressively check for the errors (i.e. first email, then valid, then password input, then password) so it looks cleaner.